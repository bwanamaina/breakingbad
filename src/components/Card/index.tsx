import React from 'react';

import Spinner from '../Spinner';
import Item from './Item';
import './style.css';

interface ICard {
  items: Array<Character>;
  isLoading: boolean;
}

const Card: React.FC<ICard> = ({ items, isLoading }) => {
  return isLoading ? (
    <Spinner />
  ) : (
    <section className="cards">
      {items &&
        items.length > 0 &&
        items.map((item) => <Item key={item.char_id} item={item}></Item>)}
    </section>
  );
};

export default Card;
