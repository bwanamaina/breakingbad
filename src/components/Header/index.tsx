import React from 'react';

import logo from '../../img/logo.png';
import './style.css';

const Header = () => {
  return (
    <header className="center">
      <img src={logo} alt="breaking bad" />
    </header>
  );
};

export default Header;
