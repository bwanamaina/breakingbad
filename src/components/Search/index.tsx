import React from 'react';

import './style.css';
interface ISearch {
  searchQuery: (query: string) => void;
}

const Search: React.FC<ISearch> = ({ searchQuery }) => {
  const [searchText, setSearchText] = React.useState('');

  return (
    <section className="search">
      <form>
        <input
          type="text"
          placeholder="Search characters"
          value={searchText}
          onChange={(e) => {
            const { value } = e.target;
            setSearchText(value);
            searchQuery(value);
          }}
          autoFocus
        />
      </form>
    </section>
  );
};

export default Search;
