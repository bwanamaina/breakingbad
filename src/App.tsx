import React from 'react';
import axios from 'axios';

import Header from './components/Header';
import Search from './components/Search';
import Card from './components/Card';

const App = () => {
  const [items, setItems] = React.useState([]);
  const [isLoading, setIsLoading] = React.useState(true);
  const [searchText, setSearchText] = React.useState('');

  React.useEffect(() => {
    (async () => {
      try {
        const { data } = await axios(
          `https://www.breakingbadapi.com/api/characters?name=${searchText}`
        );
        setItems(data);
        setIsLoading(false);
      } catch (error) {
        setIsLoading(false);
      }
    })();
  }, [searchText]);

  return (
    <div className='container'>
      <Header />
      <Search searchQuery={(query) => setSearchText(query)} />
      <Card isLoading={isLoading} items={items} />
    </div>
  );
};

export default App;
