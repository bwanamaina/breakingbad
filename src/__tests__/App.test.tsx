import React from 'react';
import { shallow, mount } from 'enzyme';

import App from '../App';
import Header from '../components/Header';
import Card from '../components/Card';
import Search from '../components/Search';

describe('<App/> Component', () => {
  let component: any;

  beforeEach(() => {
    component = shallow(<App />);
  });

  test('renders <App/> component', () => {
    expect(component).toMatchSnapshot();
  });

  test('Renders <Header/> component', () => {
    expect(component.contains(<Header />)).toBe(true);
  });

  test('Renders <Card/> component', () => {
    expect(component.find(Card)).toBeTruthy();
  });

  test('Renders <Search/> component', () => {
    expect(component.find(Search)).toBeTruthy();
  });

  it('check number of cards rendered', () => {
    const isLoading = false;
    const items: Array<Character> = [
      {
        appearance: [3, 4],
        birthday: 'Unknown',
        category: 'Breaking Bad',
        char_id: 51,
        img:
          'https://vignette.wikia.nocookie.net/breakingbad/images/8/81/Group_Leader_Brown_Shirt_-_Kafkaesque.png/revision/latest?cb=20131025095333',
        name: 'Group Leader',
        nickname: 'Counselor',
        occupation: ['Drug & Alcohol Rehabilitation Counselor'],
        portrayed: 'Jere Burns',
        status: 'Alive',
      },
    ];
    const card = mount(<Card isLoading={isLoading} items={items} />);
    expect(card.find('div.card').length).toEqual(1);
  });
});
