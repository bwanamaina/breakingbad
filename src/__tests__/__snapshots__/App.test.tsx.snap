// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> Component renders <App/> component 1`] = `ShallowWrapper {}`;
